nanoFramework.Json.Test/JsonUnitTests.cs

```diff
@@ -1384,6 +1384,34 @@ public void DeserializeObjectWithStringContainingNonAsciiChars()
             Assert.Equal(input.Value, result.Value);
         }
 
+        [TestMethod]
+        public void DeserializeObjectWithJsonIgnoreAttribute()
+        {
+            OutputHelper.WriteLine("Starting JsonIgnore Test...");
+            Json.Configuration.Settings.UseIgnoreAttribute = true;
+            OutputHelper.WriteLine("UseIgnoreAttribute enabled.");
+
+            var testObject = JsonIgnoreTestClass.CreateTestClass();
+            var jsonString = JsonSerializer.SerializeObject(testObject);
+            OutputHelper.WriteLine("After serialize.");
+            JsonIgnoreTestClass dserResult = JsonConvert.DeserializeObject(jsonString, typeof(JsonIgnoreTestClass)) as JsonIgnoreTestClass;
+            OutputHelper.WriteLine("After deserialize.");
+
+            //test serialize and deserialize
+            bool jsonSuccess = testObject.IsEqual(dserResult);
+            Assert.IsTrue(jsonSuccess);
+            OutputHelper.WriteLine("Serialization/Deserialization was " + (jsonSuccess ? "" : "NOT ") + "successful.");
```

Comment:
If you arrive here, then it's always successful. So no need to test the value of jsonSuccess

Reply:
> Looks great to me now. So all up, as a summary, it seems we are good on performance side and you even managed to optimize a bit the existing code. Correct?

I think so, as long as my caching solution is acceptable. I will get to those last few fixes you suggested as soon as I can. The one thing I didn't do is run the performance tests with the code before I touched it. I can do that as well and provide my findings when I do my next push.